### PR TITLE
Feature: enable reports

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Automate sending reports via email](https://tools.hmcts.net/jira/browse/BPS-466)

### Change description ###

**NOTE**: based on #597

Secret created in available key vaults:

```bash
INFO: Found secret reports-recipients in environment saat
INFO: Found secret reports-recipients in environment sprod
INFO: Found secret reports-recipients in environment demo
INFO: Found secret reports-recipients in environment aat
INFO: Found secret reports-recipients in environment prod
```

`prod` contains all listed in the ticket, others only bulk scan one with minor modification. There has to be some value. Reports will be and shall be enabled only for `prod`.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -179,6 +185,21 @@ data "azurerm_key_vault_secret" "s2s_secret" {
   vault_uri = "${local.s2s_vault_url}"
 }
 
+data "azurerm_key_vault_secret" "reports_recipients" {
+  name      = "reports-recipients"
+  vault_uri = "${data.azurerm_key_vault.key_vault.vault_uri}"
+}
+
+data "azurerm_key_vault_secret" "smtp_username" {
+  name      = "reports-email-username"
```
Comment (nitinprabhu) [inline infrastructure/main.tf]: make sure this secret is added to demo vault too (same as aat vault)

File: infrastructure/main.tf [hunk 2]
```diff
@@ -179,6 +185,21 @@ data "azurerm_key_vault_secret" "s2s_secret" {
...
+  vault_uri = "${data.azurerm_key_vault.key_vault.vault_uri}"
+}
+
+data "azurerm_key_vault_secret" "smtp_password" {
+  name      = "reports-email-password"
```
Comment (nitinprabhu) [inline infrastructure/main.tf]: same here as above

File: infrastructure/variables.tf [hunk 3]
```diff
@@ -117,6 +117,20 @@ variable "delete_rejected_files_cron" {
...
+
+# region reports
+variable "smtp_host" {
+  type        = "string"
+  default     = "false"
+  description = "SMTP host for sending out reports via JavaMailSender"
+}
+
+variable "reports_cron" {
+  default = "0 0 18 * * *"
```
Comment (nitinprabhu) [inline infrastructure/variables.tf]: comment would be nice what frequency it's running with :)

File: src/integrationTest/resources/application.properties [hunk 4]
```diff
@@ -28,3 +28,5 @@ user.timezone=GMT
 
 # no communication with Service Bus
 spring.profiles.active=nosb
+
+reports.recipients=integration@test
```
Comment (nitinprabhu) [inline src/integrationTest/resources/application.properties]: may be use a proper email format ? something like` integration@fake.com`

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSender.java [hunk 5]
```diff
@@ -47,6 +49,8 @@ public ReportSender(
     }
     // endregion
 
+    @Scheduled(cron = "${reports.cron:-}")
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSender.java]: can we please remove the default value from here and configure the prop in app yaml. Makes it consistent with what we have in other places.

File: infrastructure/demo.tfvars [hunk 6]
```diff
@@ -9,3 +9,5 @@ orchestrator_notifications_task_delay = "3000"
 api_gateway_test_valid_certificate_thumbprint = "C4784AD48B4B99F427D6B56FB38184D0E6457744"
 blob_signature_verification_key_file = "nonprod_public_key.der"
 allowed_client_certificate_thumbprints = ["4AF638E82FBD9959A5B8286C673360AC2C2E7053"]
+
+smtp_host = "smtp.office365.com"
```
Comment (nitinprabhu) [inline infrastructure/demo.tfvars]: Does this determine if the reports are enabled or not. Is there any other way we can do this else let's add a comment here

**Comment by doncem** [comment]
closing as people are confusing which pr is first in line

File: src/integrationTest/resources/application.properties (hunk 4 above)
Comment (doncem) [reply src/integrationTest/resources/application.properties]: don't want to use domain fully. it is valid to assign partial domain and java mail sender allows it. it's like having you local machine setup with `username@localhost`

File: infrastructure/demo.tfvars (hunk 6 above)
Comment (doncem) [reply infrastructure/demo.tfvars]: Same as feign clients are conditioned

**Pull request closed without merging**